### PR TITLE
Broker service locator improvements

### PR DESCRIPTION
**Pull request by koderi-dp**
Issue #51

[https://github.com/loekd/ServiceFabric.PubSubActors/issues/51](url)

### CODE REVIEW

File: ServiceFabric.PubSubActors/Helpers/BrokerServiceLocator.cs [hunk 1]
```diff
@@ -12,6 +12,7 @@ public class BrokerServiceLocator : IBrokerServiceLocator
     {
         private static ServicePartitionList _cachedPartitions;
         private readonly FabricClient _fabricClient;
+        private const string _borkerName = nameof(BrokerService);
```
Comment (loekd) [inline ServiceFabric.PubSubActors/Helpers/BrokerServiceLocator.cs]: small typo in field name

File: ServiceFabric.PubSubActors/Helpers/BrokerServiceLocator.cs [hunk 2]
```diff
@@ -41,25 +42,61 @@ public BrokerServiceLocator(bool useRemotingV2 = false)
...
-                var property = await fc.PropertyManager.GetPropertyAsync(new Uri(activationContext.ApplicationName), nameof(BrokerService));
-                if (property == null) return null;
-                string value = property.GetValue<string>();
-                return new Uri(value);
+                var property = await GetBrokerPropertyOrNull(activationContext.ApplicationName);
+                
+                if (property == null)
+                {
+                    // try to find broker name in other application types
+                    var apps = await _fabricClient.QueryManager.GetApplicationListAsync();
```
Comment (loekd) [inline ServiceFabric.PubSubActors/Helpers/BrokerServiceLocator.cs]: this should use the paging feature

**Comment by loekd** [comment]
@danijel-peric  Thank you for this contribution! Can you please submit the broker service locator changes separately first? And please have a look at the rebase conflicts.

**Review by loekd — CHANGES_REQUESTED** [review]

File: ServiceFabric.PubSubActors/Helpers/BrokerServiceLocator.cs (hunk 2 above)
Comment (koderi-dp) [reply ServiceFabric.PubSubActors/Helpers/BrokerServiceLocator.cs]: didn't know for paging feature, thanks, implemented in last change

File: ServiceFabric.PubSubActors/Helpers/BrokerServiceLocator.cs (hunk 1 above)
Comment (koderi-dp) [reply ServiceFabric.PubSubActors/Helpers/BrokerServiceLocator.cs]: fixed in last change

**Pull request merged**